neural_lam/utils.py

```diff
@@ -221,15 +221,51 @@ def make_mlp(blueprint, layer_norm=True):
     return nn.Sequential(*layers)
 
 
+def has_working_latex():
```

Comment:
This is a great check! As this is a bit heavier to run now, and since it should always return the same thing during the course of the program running, I think we should cache the output of this function.

```suggestion
@cache
def has_working_latex():
```

(also requires importing [`functools.cache`](https://docs.python.org/dev/library/functools.html#functools.cache))

Reply:
Great! I think you might have to import it as well (`from functools import cache` at the top of the file) for it to work. Please try it out.